### PR TITLE
[Sluggable] Add datetime_immutable as valid type

### PR DESCRIPTION
**Pull request by pbreteche**
Slug fields already allow datetime and datetimez, all Doctrine date type variations that instanciate \DateTime and \DateTimeImmutable objects should work either.

#### Added
- Add support for DateTimeImmutable fields

### CODE REVIEW

**Comment by franmomu** [comment]
Thanks @pbreteche! Can you please add a test? I think we don't have any using a `\DateTimeInterface` to generate a slug and that would be awesome and please add an entry to the `CHANGELOG.md` file.

**Comment by pbreteche** [comment]
@franmomu as you said, I don't find any tests with `\DateTimeInterface`. I hope this is ok with this project coding style.

File: tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDate.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    #[ORM\Column(name: 'created_at', type: Types::DATE_MUTABLE)]
+    private $createdAt;
+
+    /**
+     * @var string|null
+     *
+     * @Gedmo\Slug(separator="-", updatable=true, fields={"title", "createdAt"}, dateFormat="Y-m-d")
+     * @ORM\Column(name="slug", type="string", length=64, unique=true)
+     */
+    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormate: 'Y-m-d')]
```
Comment (dritter) [inline tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDate.php:59]:
```suggestion
    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormat: 'Y-m-d')]
```

File: tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateImmutable.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    #[ORM\Column(name: 'created_at', type: Types::DATE_IMMUTABLE)]
+    private $createdAt;
+
+    /**
+     * @var string|null
+     *
+     * @Gedmo\Slug(separator="-", updatable=true, fields={"title", "createdAt"}, dateFormat="Y-m-d")
+     * @ORM\Column(name="slug", type="string", length=64, unique=true)
+     */
+    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormate: 'Y-m-d')]
```
Comment (dritter) [inline tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateImmutable.php:59]:
```suggestion
    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormat: 'Y-m-d')]
```

File: tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTime.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    #[ORM\Column(name: 'created_at', type: Types::DATETIME_MUTABLE)]
+    private $createdAt;
+
+    /**
+     * @var string|null
+     *
+     * @Gedmo\Slug(separator="-", updatable=true, fields={"title", "createdAt"}, dateFormat="Y-m-d")
+     * @ORM\Column(name="slug", type="string", length=64, unique=true)
+     */
+    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormate: 'Y-m-d')]
```
Comment (dritter) [inline tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTime.php:59]:
```suggestion
    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormat: 'Y-m-d')]
```

File: tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTimeImmutable.php [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    #[ORM\Column(name: 'created_at', type: Types::DATETIME_IMMUTABLE)]
+    private $createdAt;
+
+    /**
+     * @var string|null
+     *
+     * @Gedmo\Slug(separator="-", updatable=true, fields={"title", "createdAt"}, dateFormat="Y-m-d")
+     * @ORM\Column(name="slug", type="string", length=64, unique=true)
+     */
+    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormate: 'Y-m-d')]
```
Comment (dritter) [inline tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTimeImmutable.php:59]:
```suggestion
    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormat: 'Y-m-d')]
```

File: tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTimeTz.php [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    #[ORM\Column(name: 'created_at', type: Types::DATETIMETZ_MUTABLE)]
+    private $createdAt;
+
+    /**
+     * @var string|null
+     *
+     * @Gedmo\Slug(separator="-", updatable=true, fields={"title", "createdAt"}, dateFormat="Y-m-d")
+     * @ORM\Column(name="slug", type="string", length=64, unique=true)
+     */
+    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormate: 'Y-m-d')]
```
Comment (dritter) [inline tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTimeTz.php:59]:
```suggestion
    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormat: 'Y-m-d')]
```

File: tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTimeTzImmutable.php [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    #[ORM\Column(name: 'created_at', type: Types::DATETIMETZ_IMMUTABLE)]
+    private $createdAt;
+
+    /**
+     * @var string|null
+     *
+     * @Gedmo\Slug(separator="-", updatable=true, fields={"title", "createdAt"}, dateFormat="Y-m-d")
+     * @ORM\Column(name="slug", type="string", length=64, unique=true)
+     */
+    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormate: 'Y-m-d')]
```
Comment (dritter) [inline tests/Gedmo/Sluggable/Fixture/DateTimeTypes/ArticleDateTimeTzImmutable.php:59]:
```suggestion
    #[Gedmo\Slug(separator: '-', updatable: true, fields: ['title', 'createdAt'], dateFormat: 'Y-m-d')]
```

**Review by dritter — CHANGES_REQUESTED** [review]
There seems to be a typo in the attribute name..

File: CHANGELOG.md [hunk 7]
```diff
@@ -18,6 +18,8 @@ a release.
 ---
 
 ## [Unreleased]
+#### Added
+- Sluggable: Add support for DateTimeImmutable fields
 
 #### Added
```
Comment (phansys) [inline CHANGELOG.md:24]:
```suggestion
#### Added
- Sluggable: Add support for DateTimeImmutable fields
```

**Review by phansys — APPROVED** [review]

**Comment by phansys** [comment]
Thank you @pbreteche and @franmomu!

**Pull request merged**